src/components/PDFView/index.tsx

```diff
@@ -0,0 +1,133 @@
+import 'core-js/features/array/at';
+import React, {memo, useCallback, useEffect, useRef, useState} from 'react';
+import {PDFPreviewer} from 'react-fast-pdf';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import PressableWithoutFeedback from '@components/Pressable/PressableWithoutFeedback';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import variables from '@styles/variables';
+import * as CanvasSize from '@userActions/CanvasSize';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import PDFPasswordForm from './PDFPasswordForm';
+import type {PDFViewOnyxProps, PDFViewProps} from './types';
+
+function PDFView({onToggleKeyboard, fileName, onPress, isFocused, sourceURL, errorLabelStyles, maxCanvasArea, maxCanvasHeight, maxCanvasWidth, style}: PDFViewProps) {
+    const [isKeyboardOpen, setIsKeyboardOpen] = useState(false);
+    const styles = useThemeStyles();
+    const {windowHeight, isSmallScreenWidth} = useWindowDimensions();
+    const prevWindowHeight = useRef(windowHeight);
+    const {translate} = useLocalize();
+
+    /**
+     * On small screens notify parent that the keyboard has opened or closed.
+     *
+     * @param {Boolean} isKeyboardOpen True if keyboard is open
+     */
+    const toggleKeyboardOnSmallScreens = useCallback(
+        (isKBOpen: boolean) => {
+            if (!isSmallScreenWidth) {
+                return;
+            }
+            setIsKeyboardOpen(isKBOpen);
+            onToggleKeyboard?.(isKeyboardOpen);
+        },
+        [isKeyboardOpen, isSmallScreenWidth, onToggleKeyboard],
+    );
+
+    /**
+     * Verify that the canvas limits have been calculated already, if not calculate them and put them in Onyx
+     */
+    const retrieveCanvasLimits = () => {
+        if (!maxCanvasArea) {
+            CanvasSize.retrieveMaxCanvasArea();
+        }
+
+        if (!maxCanvasHeight) {
+            CanvasSize.retrieveMaxCanvasHeight();
+        }
+
+        if (!maxCanvasWidth) {
+            CanvasSize.retrieveMaxCanvasWidth();
+        }
+    };
+
+    useEffect(() => {
+        retrieveCanvasLimits();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    // shouldComponentUpdate(nextProps, nextState) {
+    //     return !_.isEqual(state, nextState) || !_.isEqual(props, nextProps);
+    // }
+
+    useEffect(() => {
+        if (!isKeyboardOpen && windowHeight < prevWindowHeight.current - 100) {
```

Comment:
```suggestion
        // Use window height changes to toggle the keyboard. To maintain keyboard state
        // on all platforms we also use focus/blur events. So we need to make sure here
        // that we avoid redundant keyboard toggling.
        // Minus 100px is needed to make sure that when the internet connection is
        // disabled in android chrome and a small 'No internet connection' text box appears,
        // we do not take it as a sign to open the keyboard
        if (!isKeyboardOpen && windowHeight < prevWindowHeight.current - 100) {
```

Reply:
@fabioh8010 you were inspired when you wrote this comment, me gusta 😄 

Thank you for the info!